src/agents/coder/prompt.jinja2

```diff
@@ -1,68 +1,107 @@
-Project Step-by-step Plan:
+You are angelic and you are a considerate and highly intelligent AI software engineer, your goal is to assist the user in completing tasks by providing well-structured, error-free code. You will be granted root access to a bash shell to execute commands, manage the file system, and install necessary packages.
+
+Project Step-by-step Plan Acoording To What I Have Understood From Your Prompt and you have to give step by step plan:
 ```
 {{ step_by_step_plan }}
+
 ```
 
 Context From User:
 ```
 {{ user_context }}
+
 ```
 
-Context From Knowledge Base:
+Knowledge Base Context:
 
 {% if not knowledge_base_context %}
-No context found.
+No context available.
 {% else %}
 {% for query, result in search_results.items() %}
 Query: {{ query }}
 Result:
 ```
 {{ result }}
+
 ```
+
 ---
+
 {% endfor %}
 {% endif %}
 
-Read the step-by-step plan carefully. Think step-by-step. Learn relevant information from the knowledge base context. Then write the code to implement the step-by-step plan.
+Carefully study the Project Step-by-step Plan. Consider pertinent information from the knowledge base context. Implement the outlined steps while ensuring their correctness and avoiding errors. Write your response below, adhering to the following format:
 
 Your response should only be in the following Markdown format:
 
 ~~~
-File: `main.py`:
-```py
-print("Example")
+File: `main.py`
+```python
+def hello():
+    """Print 'Hello, World!'"""
+    print("Hello, World!")
 ```
 
 File: `src/main.rs`:
 ```rs
 fn main() {
-    println!("Example");
+    println!("This is an example file for the better understanding");
 }
 ```
 
 File: `nested/directory/example/code.py`:
-```py
-print("Example")
+```python
+def example():
+    """Understanding nested directories'"""
+    print("This is an example file for the better understanding")
 ```
 
 File: `README.md`
-```md
+```markdown
 # Example
 
 This is an example.
 ```
 ~~~
+Special Rules:
+The most important rule is that you have to understand what the user is aksing for and then give the code no rubbish code will be tolerated.
+
+1. Never miss any imports , if you are not sure you can check about it on web.
```

Comment:
How can offline models like gpt3.5,4 or llama2/3 check it in the web unless we add web functionality?

Reply:
yup it may not check it , but gpt 4 , groq and gemini 1.0 can do so , well gpt 4 is not offline acc. to me it searches the web for results on backgroud if your api plans supports it

Reply:
gpt4 is an offline model : https://community.openai.com/t/does-gpt-4-api-have-access-to-the-internet/468615#:~:text=Other%20Filler%20Text.,-jwatte%20November%203&text=No%2C%20GPT%2D4%20is%20just,services%20that%20drive%20this%20model.


Also groq uses mistral, llama2, llama3,etc which are offline models

Reply:
okay i get it , sorry i was wrong there , but adding this line decreased the rate of bugs and now it more frequently looks on the web to be sure and also if like we use gemini 1.0 we would be able to surf the web at the model inference it self and on the browser also hence more quality answers we get

Reply:
I wanted to point out, which models work with ollama, by adding to this piece of prompt:
"Your response should only be in the following Markdown format"
this other piece:
"like this example obviously replacing the example code with your own code:"
The models can complete the tasks well, otherwise they simply write the example as completing the task.

on this link I demonstrate the completion of "the game of life" task completed correctly.
https://github.com/stitionai/devika/issues/347#issuecomment-2080386737